docs/DeveloperGuide.md

```diff
@@ -1,38 +1,254 @@
 # Developer Guide
 
-## Acknowledgements
+## Introduction
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+Budget Buddy aims to help those who want a Command Line Interface App that is a one-stop solution to keep track of their
+expenses, help them with budgeting and also be help them to visualise what they have spent.
 
-## Design & implementation
+Overspending and the lack of savings is a significant problem found in most fresh graduates. We want to create tools
+necessary for them so that they can realize their actual spending and achieve financial freedom in the future.
+
+## Setting up the project
+
+## Design
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Main Component
+
+![MainSeq.png](uml/MainSequence.png)
+
+In the main (Duke) class, the main method first calls startApplication() which initalizes the Log Manager for logging (
+disabled if in production stage) and Ui object for printing output messages. After initialized, it will then attempt to
+import the user's data from `data.json`. If there is no file, it will initalize a blank data. If importing fails, the
+application will print an error message and exit with code 1. Once done, it will greet the user with a message.
+
+The application then runs an infinite loop to take in and execute commands, until when the inputs `exit` to exit the
+application. After which, it will output a bye message and exits the application safely.
+
+### Command Component
+
+![Command.png](uml/Command.png)
+
+The command consists of three components: Enum CommandEnum, Abstract class Command and Class CommandParser. Each of them
+plays a role in retrieving the commands input by the user and redirecting to the correct components.
+
+#### CommandEnum
+
+The CommandEnum Enum contains all the commands that are available in the application.
+These are the only commands that can be input in the command line.
+
+#### Command
+
+The abstract class Command extends to several commands (eg `BudgetCommand`, `DepositCommand`, `ExpenseCommand`) and contains abstract methods execute() and isExit().
+The execute() method is executed from main, where it will go to the action class (eg `BudgetAction`) and run the requested action such as `add`.
+After execute() from main, isExit() controls the termination of BudgetBuddy by returning a boolean.
+
+#### CommandParser
+
+![CommandParserSequence.png](uml/CommandParserSequence.png)
```

Comment:
<img width="246" alt="image" src="https://user-images.githubusercontent.com/81511355/228761346-93564788-6004-42fd-9a03-209233f50a03.png">

The X mark is overlapped in the activation bar. Should be after the activation bar